rest/data/pod.go

```diff
@@ -42,3 +83,48 @@ func (c *MockPodConnector) CheckPodSecret(id, secret string) error {
 	}
 	return errors.New("pod does not exist")
 }
+
+func validatePod(p restModel.APICreatePod) (*pod.Pod, error) {
+	if p.Image == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("Invalid API input: missing image"),
+		}
+	}
+
+	for _, envVar := range p.EnvVars {
+		if envVar.SecretOpts == nil {
+			if envVar.Name == nil {
+				return nil, gimlet.ErrorResponse{
+					StatusCode: http.StatusBadRequest,
+					Message:    fmt.Sprintf("Invalid API input: missing environment variable name"),
+				}
+			}
+		} else {
+			if envVar.SecretOpts.Name == nil {
```

Comment:
I feel like this validation is excessively complicated - you have either a regular environment variable or a set of secret options which are completely separate and store the same parameters. Can we simplify the environment variable handling so that the environment variable name is both the name of the environment variable and the name of the secret? The DB pod already makes the assumption that the name of the stored secret is the same as the environment variable name.

Like this:
```go
type APIPodEnvVar struct {
    Name *string
    Value *string
    Secret *bool
}
```
where the secret boolean will determine if it's supposed to be stored as a secret or as a plaintext environment variable.